src/contexts/UrlContextProvider.tsx

```diff
@@ -204,13 +205,18 @@ const getWindowUrlHashParams = () => {
     );
     const hashParams = new URLSearchParams(window.location.hash.substring(1));
 
-    const logEventNum = hashParams.get(HASH_PARAM_NAMES.LOG_EVENT_NUM);
-    if (null !== logEventNum) {
-        const parsed = Number(logEventNum);
-        urlHashParams[HASH_PARAM_NAMES.LOG_EVENT_NUM] = Number.isNaN(parsed) ?
-            null :
-            parsed;
-    }
+    const checkAndSetHashParam = (hashParamName: keyof UrlHashParams) => {
+        const hashParam = hashParams.get(hashParamName);
+        if (null !== hashParam) {
+            const parsed = Number(hashParam);
+            urlHashParams[hashParamName] = Number.isNaN(parsed) ?
+                null :
+                parsed;
+        }
+    };
+
+    checkAndSetHashParam(HASH_PARAM_NAMES.LOG_EVENT_NUM);
+    checkAndSetHashParam(HASH_PARAM_NAMES.TIMESTAMP);
```

Comment:
Would this be more concise and avoid the nested function declaration? (haven't tested
```suggestion
    const numberHashParamNames = [HASH_PARAM_NAMES.LOG_EVENT_NUM, HASH_PARAM_NAMES.TIMESTAMP];
    for (const paramName of numberHashParamNames) {
        const hashParam = hashParams.get(paramName);
        if (hashParam !== null) {
            const parsed = Number(hashParam);
            urlHashParams[paramName] = Number.isNaN(parsed) ? null : parsed;
        }
    }
```